### PR TITLE
Footer implementation

### LINKED ISSUE #22: Footer component
**Epic** https://github.com/x-team/auto/issues/573

---

Create new component.

screen shot 2017-11-07 at 10 46 44 am

### Dependencies

- FooterList component https://github.com/x-team/auto-ui/issues/23
- FooterBrands component https://github.com/x-team/auto-ui/issues/24
- HorizontalRuler component https://github.com/x-team/auto-ui/issues/21 

### Props

```
children: node|array,
```

### Behaviour

As pointed out in https://github.com/x-team/auto/issues/573

> Please include the footer from x-team.com, available via the x-team-com-new repo (templates/footer.js). Request access from @joshwnj if you don't have access to that repo.

- `children` should get FooterList and FooterBrands

Example usage:

```
<Footer>
  <FooterList title="list one" items={list1} />
  <FooterList title="list two" items={list2} />
  <FooterList title="list tree" items={list3} />
  <FooterList title="list four" items={list4} />
  <FooterBrands brands={listOfBrands} />
</Footer>
```

### LINKED ISSUE #23: FooterList component
**Epic** https://github.com/x-team/auto/issues/573

---

Create new component.

screen shot 2017-11-07 at 10 49 58 am

### Props

```
title: string,
items: [
  {
    label: string,
    url: string,
  }
]
```

### Behaviour

None specific behaviour.

### LINKED ISSUE #24: FooterBrands component
**Epic** https://github.com/x-team/auto/issues/573

---

Create new component.

screen shot 2017-11-07 at 10 52 08 am

### Props

```
brands: [
  {
    title: string,
    image: string,
    url: string,
  }
]
```

### Behaviour

None specific behaviour.

### PR DESCRIPTION
**Pull request by skyronic**
Closes #22, closes #23, closes #24

### CODE REVIEW

File: src/components/Footer/FooterBrands.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    text-transform: uppercase;
+    opacity: .5;
+    margin: 0;
+  `,
+    typo.family.base
+  )
+}
+
+export default class FooterBrands extends PureComponent<Props> {
+  static defaultProps = {}
```
Comment (nicksp) [inline src/components/Footer/FooterBrands.js]: No need to define empty defaults

File: src/components/Footer/FooterList.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    font-family: "Source Sans Pro", "Helvetica Neue", Helvetica, Arial, sans-serif;
+    font-size: 1.25rem;
+    text-transform: uppercase;
+    color: ${theme.black}
+    margin: 0 0 2em;
+  `)
+}
+
+export default class FooterList extends PureComponent<Props> {
+  static defaultProps = {}
```
Comment (nicksp) [inline src/components/Footer/FooterList.js]: same here, useless LOC

File: src/components/HeaderBar.js [hunk 3]
```diff
@@ -17,41 +18,13 @@ const cx = {
...
-      <path
-        d='M 185.9194,70.171973 259.8985,142.67307 331.4976,72.501145 257.5202,4.8828125e-5 185.9194,70.171973'
-        fill='#231f20'
-      />
-    </svg>
-  )
-}
-
 export default class HeaderBar extends PureComponent<Props> {
   static defaultProps = {}
```
Comment (nicksp) [inline src/components/HeaderBar.js]: remove this.

File: src/components/Footer/Footer.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    opacity: 0.5;
+    line-height: 0;
+    color: ${theme.black}
+  `,
+    typo.family.base
+  )
+}
+
+export default class Footer extends PureComponent<Props> {
+  static defaultProps = {}
```
Comment (nicksp) [inline src/components/Footer/Footer.js]: remove this.

**Comment by skyronic** [comment]
@nicksp fixed the `defaultProps`

**Comment by joshwnj** [comment]
Looks good, thanks @skyronic

**Pull request merged**